### PR TITLE
Started working on MovieDetail screen UI

### PR DESCRIPTION
**Pull request by amitkumar693**
- Created UI for movie detail screen for Cast and Crew and Plot summary.

Simulator Screen Shot - iPhone 11 Pro Max - 2020-05-04 at 22 58 19

### CODE REVIEW

**Comment by NilaakashSingh** [comment]
As discussed please comment once feedback is pushed

File: MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+struct CastAndCrewView: View {
+    let castAndCrews: [CastAndCrew]
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("Cast and Crew")
+                .padding(.all)
+            ScrollView(.horizontal, showsIndicators: false) {
+                HStack(alignment: .top, spacing: 10) {
+                    ForEach(castAndCrews) { castAndCrew in
```
Comment (swapnil-salunke) [inline MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: Can we use map here?

File: MovieDemoApp/Views/Movie Detail/Movie Image and Ratings/ImageAndRatingsView.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        ZStack(alignment: .bottomTrailing) {
+            MoviewImageView(image: image)
+            MovieRatingsView()
+        }
+
+    }
+}
+
+// View Modifier for custom side corner radius
+struct CornerRadiusStyle: ViewModifier {
```
Comment (swapnil-salunke) [inline MovieDemoApp/Views/Movie Detail/Movie Image and Ratings/ImageAndRatingsView.swift]: could we not use rounded rectangle object directly for this purposes? Any specific reason we used a view modifier?

Comment (amitkumar693) [reply MovieDemoApp/Views/Movie Detail/Movie Image and Ratings/ImageAndRatingsView.swift]: Rounded rectangle does not allow to add corner radius for specific size.
Using this modifier I was able to add corner radius to any specific side.

File: MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift (hunk 1 above)
Comment (amitkumar693) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: I tried using :

    `castAndCrews.forEach( CastAndCrewSingleView(castAndCrew: $0 ))`

But I am getting error: 
> Static member 'leading' cannot be used on instance of type 'HorizontalAlignment'

Comment (NilaakashSingh) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: you can't use forEach inside body

Comment (NilaakashSingh) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: you can't use .forEach inside body
ForEach is a swift UI keyword.
ForEach != .forEach

Comment (swapnil-salunke) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: @NilaakashSingh so this is specific to Swift UI and we cannot use .map types of functions here?

Comment (amitkumar693) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: In Apple Doc: 

ForEach is an struct:

     /// A structure that computes views on demand from an underlying collection of
     /// of identified data.
     @available(iOS 13.0, OSX 10.15, tvOS 13.0, watchOS 6.0, *)
     public struct ForEach<Data, ID, Content> where Data : RandomAccessCollection, ID : 
     Hashable {

        /// The collection of underlying identified data.
        public var data: Data

        /// A function that can be used to generate content on demand given
        /// underlying data.
        public var content: (Data.Element) -> Content
    }

and .forEach is closure: 

    /// - Parameter body: A closure that takes an element of the sequence as a
    ///   parameter.
    @inlinable public func forEach(_ body: (Self.Element) throws -> Void) rethrows

Comment (NilaakashSingh) [reply MovieDemoApp/Views/Movie Detail/Cast And Crew/CastAndCrewView.swift]: yeah @swapnil-salunke we can't use .map directly with UI components of SwiftUI. On the other hand, we can do it via the Combine model. We can keep it as it is now.

**Pull request merged**